tests/unittests/test_datasource/test_azure.py

```diff
@@ -2270,6 +2418,220 @@ def test_reprovision_calls__poll_imds(self, _poll_imds, isfile):
         _poll_imds.assert_called_with()
 
 
+class TestPreprovisioningHotAttachNics(CiTestCase):
+
+    def setUp(self):
+        super(TestPreprovisioningHotAttachNics, self).setUp()
+        self.tmp = self.tmp_dir()
+        self.waagent_d = self.tmp_path('/var/lib/waagent', self.tmp)
+        self.paths = helpers.Paths({'cloud_dir': self.tmp})
+        dsaz.BUILTIN_DS_CONFIG['data_dir'] = self.waagent_d
+        self.paths = helpers.Paths({'cloud_dir': self.tmp})
+
+    @mock.patch('cloudinit.sources.helpers.netlink.wait_for_nic_detach_event')
```

Comment:
```suggestion
    @mock.patch('cloudinit.sources.helpers.netlink.wait_for_nic_detach_event', autospec=True)
```

(This may need rewrapping for line length.)